core/unix/signal.c

```diff
@@ -3530,7 +3530,14 @@ unlink_fragment_for_signal(dcontext_t *dcontext, fragment_t *f,
         LOG(THREAD, LOG_ASYNCH, 3,
             "\tunlinking outgoing for interrupted F%d\n", f->id);
         SHARED_FLAGS_RECURSIVE_LOCK(f->flags, acquire, change_linking_lock);
-        unlink_fragment_outgoing(dcontext, f);
+        // Double-check flags to ensure some other thread didn't unlink
+        // while we waited for the change_linking_lock.
+        if (TEST(FRAG_LINKED_OUTGOING, f->flags)) {
+            unlink_fragment_outgoing(dcontext, f);
+            // TODO(during review): should we set `changed`? the code below that
```

Comment:
I assume you mean for the else case: no, changed should not be set there, since if set then the caller tracks this fragment and relinks it upon delivering the signal, when we want whoever else unlinked it to do that and not us.

Reply:
Thanks! That's what I thought, just wanted to make sure.